avocado/utils/network/interfaces.py

```diff
@@ -144,6 +144,18 @@ def bring_up(self):
         except Exception as ex:
             raise NWException("Failed to bring up: %s" % ex)
 
+    def get_admin_link_state(self):
+        """Get admin link state.
+        :return: admin link state
+        """
+        cmd = "ip link show {}".format(self.name)
+        try:
+            output = run_command(cmd, self.host, sudo=True)
+            state = output.splitlines()[0].split('>')[0].split(',')[-1]
```

Comment:
we are already using `-j` (json output) option when using `_get_interface_details()`. You could avoid the splits and this index syntax.

Also `_get_interface_details()` is already handling some exceptions and looks like it has the information that you need, that would make your method much cleaner:

```python
     def get_admin_link_state(self):
          try:
              return self._get_interface_details().get('flags')[2]
          except (NWException, IndexError):
              raise NWException("Could not get admin link state.")
```